### ISSUE TITLE
java.lang.NumberFormatException: For input string: "yesi"

### ISSUE THREAD

**Issue by asmotrich** [sbt/sbt#6243]
## steps

SBT 1.4.6 on a command line and invoked from IntelliJ
Add dependency to a multi-module project and run updateClassifiers. This is run by IDE to synchronize SBT and IntelliJ project structure.

## problem

```scala
sbt:pm61-reasoner> last catalogProject / updateClassifiers
[debug] not up to date. inChanged = true, force = true
[debug] Updating ProjectRef(uri("file:/orion/forge/promethium/pm61data/pm61-reasoner/"), "catalogProject")...
[error] java.lang.NumberFormatException: For input string: "yesi"
[error]         at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
[error]         at java.lang.Integer.parseInt(Integer.java:580)
[error]         at java.math.BigInteger.<init>(BigInteger.java:470)
[error]         at lmcoursier.internal.shaded.coursier.cache.CacheChecksum$$anonfun$findChecksum$1.applyOrElse(CacheChecksum.scala:22)
[error]         at lmcoursier.internal.shaded.coursier.cache.CacheChecksum$$anonfun$findChecksum$1.applyOrElse(CacheChecksum.scala:20)
[error]         at scala.collection.TraversableOnce.collectFirst(TraversableOnce.scala:172)
[error]         at scala.collection.TraversableOnce.collectFirst$(TraversableOnce.scala:159)
[error]         at scala.collection.AbstractTraversable.collectFirst(Traversable.scala:108)
[error]         at lmcoursier.internal.shaded.coursier.cache.CacheChecksum$.findChecksum(CacheChecksum.scala:20)
[error]         at lmcoursier.internal.shaded.coursier.cache.CacheChecksum$.$anonfun$parseChecksumAlternative$2(CacheChecksum.scala:31)
[error]         at scala.Option.orElse(Option.scala:447)
[error]         at lmcoursier.internal.shaded.coursier.cache.CacheChecksum$.parseChecksumAlternative(CacheChecksum.scala:30)
[error]         at lmcoursier.internal.shaded.coursier.cache.CacheChecksum$.$anonfun$parseRawChecksum$1(CacheChecksum.scala:52)
[error]         at scala.Option.orElse(Option.scala:447)
[error]         at lmcoursier.internal.shaded.coursier.cache.CacheChecksum$.parseRawChecksum(CacheChecksum.scala:52)
[error]         at lmcoursier.internal.shaded.coursier.cache.FileCache.$anonfun$validateChecksum$4(FileCache.scala:677)
[error]         at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
[error]         at scala.util.Success.$anonfun$map$1(Try.scala:255)
[error]         at scala.util.Success.map(Try.scala:213)
[error]         at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
[error]         at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
[error]         at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
[error]         at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
[error]         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[error]         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[error]         at java.lang.Thread.run(Thread.java:748)
[error] (catalogProject / updateClassifiers) java.lang.NumberFormatException: For input string: "yesi"
```

## expectation

Expect no exceptions.

## notes

This is very similar to #1425 and #5218 that were reported as fixed in SBT 1.3.3. This is currently happening on 1.4.6.

**Comment by eed3si9n** [comment]
@asmotrich Thanks for the report. Could you provide reproduction steps so we can reproduce your problem on _our computers_?

Looking at the stack trace

```
[error] at java.math.BigInteger.(BigInteger.java:470)
[error] at lmcoursier.internal.shaded.coursier.cache.CacheChecksum$$anonfun$findChecksum$1.applyOrElse(CacheChecksum.scala:22)
[error] at lmcoursier.internal.shaded.coursier.cache.CacheChecksum$$anonfun$findChecksum$1.applyOrElse(CacheChecksum.scala:20)
```

The function that's causing it seems to be from Coursier https://github.com/coursier/coursier/blob/v2.0.8/modules/cache/jvm/src/main/scala/coursier/cache/CacheChecksum.scala#L16-L23, so could you report this to coursier/coursier too please?

**Comment by asmotrich** [comment]
@eed3si9n Thank you for feedback. Making a reproducible case is difficult: this is a Spark project with very large dependency tree. And the way the error represents itself is: I add a dependency and sbt crashes, I comment it out and everything else is back to normal. This exception comes for a very specific updateClassifiers task, it is used by IntelliJ during project sync with build.sbt file. You rarely use it in daily development cycle, but it is very reproducible on a command line too.

Can you recommend a way to debug the sbt & plugin in-situ, that way I can produce more info.

**Comment by eed3si9n** [comment]
> I add a dependency and sbt crashes, I comment it out and everything else is back to normal.

Is that a public dependency? If so can I add it to `build.sbt` and reproduce the problem? If not then it's ok.

**Comment by asmotrich** [comment]
It is public: libraryDependencies += "net.sansa-stack" %% "sansa-rdf-spark" % "0.7.2-SNAPSHOT" % Compile,

you need these repositories:
resolvers += "AKSW snapshot repository" at "https://maven.aksw.org/archiva/repository/snapshots/",
resolvers += "AKSW release repository" at "https://maven.aksw.org/archiva/repository/internal/",

**Comment by asmotrich** [comment]
Sorry, you need this:
      resolvers += "AKSW snapshot repository" at "https://maven.aksw.org/archiva/repository/snapshots/",
      resolvers += "AKSW release repository" at "https://maven.aksw.org/archiva/repository/internal/",

**Comment by eed3si9n** [comment]
This was fixed in upstream by @chimmi - https://github.com/coursier/coursier/pull/1972, and released as Coursier 2.0.13.
